test/unit/tedious-test.js

```diff
@@ -23,3 +23,27 @@ exports.connection = function(test) {
 
   test.done();
 };
+
+exports.connectionDoesNotModifyPassedConfig = function(test) {
+  var config = {
+    userName: 'sa',
+    password: 'sapwd',
+    options: {
+      port: 1234,
+      cryptoCredentialsDetails: {
+        ciphers: 'RC4-MD5'
+      }
+    }
+  };
+
+  var connection = new Connection(config);
+
+  test.notStrictEqual(connection.config, config);
+  test.notStrictEqual(connection.config.options, config.options);
+
+  // Test that we did not do a deep copy of the cryptoCredentialsDetails,
+  // as we never modify that value inside tedious.
+  test.strictEqual(connection.config.options.cryptoCredentialsDetails, config.options.cryptoCredentialsDetails);
```

Comment:
My comment was not so much about Node.js modifying the object, rather the application modifying it. Something like:

```JavaScript
... setup config ...
config.cryptoCredentialsDetails = { some stuff. }
var conn1 = new Connection(config);
... stuff with conn1...

... change a few fields and reuse config....
config.server = someotherserver;
config.cryptoCredentialsDetails = { some other stuff }
var conn2 = new Connection(config);

// Now you got both conn1 and conn2 operating with cryptoCredentialsDetails from the second config.
```
It would seem like that's an easy mistake to make. Should we protect the user application (not Node.js) from Tedious changing the config, as well as protect Tedious from user application change config?